src/Identity/Startup.cs

```diff
@@ -123,7 +123,7 @@ public void ConfigureServices(IServiceCollection services)
 
                             if (context.Properties.Parameters.Count > 0 && context.Properties.Parameters.ContainsKey(SsoTokenable.TokenIdentifier))
```

Comment:
Is there a `TryGet` on `Parameters`? I would love to switch to that. You know how much I love that pattern.

Reply:
So you would prefer
```csharp
var token = context.Properties.Parameters.TryGetValue(SsoTokenable.TokenIdentifier, out var tokenValue) ? tokenValue.ToString() : "";
```

That is way worse IMO.

The issue also isn't that the key isn't in the dictionary, it's that the value is null, so in the above we'd even need null checking in the `true` response.

```csharp
var token = context.Properties.Parameters.TryGetValue(SsoTokenable.TokenIdentifier, out var tokenValue) ? tokenValue?.ToString() ?? "" : "";
```

Reply:
@MGibson1 I would probably do something like this:
```C#
if (context.Properties.Parameters.TryGetValue(SsoTokenable.TokenIdentifier, out var tokenValue))
{
   context.ProtocolMessage.Parameters.Add(SsoTokenable.TokenIdentifier, tokenValue is string token ? token : "");
}
```
because we should be able to assert that it's a string since we are setting it. But if you want to keep the `.ToString()` you can do this:

```C#
if (context.Properties.Parameters.TryGetValue(SsoTokenable.TokenIdentifier, out var tokenValue))
{
    var token = tokenValue?.ToString() ?? "";
    context.ProtocolMessage.Parameters.Add(SsoTokenable.TokenIdentifier, token);
}
```

This is the same thing as what you proposed but with searching the dictionary one less time.